tree/ntuple/v7/test/ntuple_emulated.cxx

```diff
@@ -0,0 +1,90 @@
+#include "ntuple_test.hxx"
+
+#define NTUPLE_FORK_MACROS
+#include "ntuple_fork.hxx"
+
+TEST(RNTupleEmulated, EmulatedFields)
+{
+   // Write a user-defined class to RNTuple, then "forget" about it and try to load it as an emulated record field.
+   // To achieve this "forgetting" we use fork() so that the parent process doesn't know about the class.
+
+   FileRaii fileGuard("test_ntuple_emulated_fields.root");
+
+   ExecInFork([&] {
+      // The child process writes the file and exits, but the file must be preserved to be read by the parent.
+      fileGuard.PreserveFile();
+
+      ASSERT_TRUE(gInterpreter->Declare(R"(
+struct Inner {
+   int fInt1 = 1;
+   int fInt2 = 2;
+
+   ClassDefNV(Inner, 2);
+};
+
+struct Outer {
+   int fInt1 = 1;
+   Inner fInner;
+
+   ClassDefNV(Outer, 2);
+};
+)"));
+
+      auto model = RNTupleModel::Create();
+      model->AddField(RFieldBase::Create("f", "Outer").Unwrap());
+
+      auto writer = RNTupleWriter::Recreate(std::move(model), "ntpl", fileGuard.GetPath());
+      writer->Fill();
+
+      void *ptr = writer->GetModel().GetDefaultEntry().GetPtr<void>("f").get();
+      DeclarePointer("Outer", "ptrOuter", ptr);
+
+      ProcessLine("ptrOuter->fInner.fInt1 = 71;");
+      ProcessLine("ptrOuter->fInner.fInt2 = 82;");
+      ProcessLine("ptrOuter->fInt1 = 93;");
+      writer->Fill();
+
+      // TStreamerInfo::Build will report a warning for interpreted classes (but only for members).
+      // See also https://github.com/root-project/root/issues/9371
+      ROOT::TestSupport::CheckDiagsRAII diagRAII;
+      diagRAII.optionalDiag(kWarning, "TStreamerInfo::Build", "has no streamer or dictionary",
+                            /*matchFullMessage=*/false);
+      writer.reset();
+   });
+
+   auto reader = RNTupleReader::Open("ntpl", fileGuard.GetPath());
+   const auto &desc = reader->GetDescriptor();
+
+   {
+      RNTupleDescriptor::RCreateModelOptions opts;
+      opts.fEmulateUnknownTypes = false;
+      try {
+         auto model = desc.CreateModel(opts);
+         FAIL() << "Creating a model without fEmulateUnknownTypes should fail";
+      } catch (RException ex) {
```

Comment:
change to `ROOT::RException` needs to move here from another commit, and the compiler complains:
```
catching polymorphic type ‘class ROOT::RException’ by value
```